include/convex_bodies/ballintersectconvex.h

```diff
@@ -123,13 +122,74 @@ class BallIntersectPolytope {
 
         std::pair <NT, int> polypair = P.line_positive_intersect(r, v, Ar, Av, lambda_prev);
         std::pair <NT, int> ball_lambda = B.line_positive_intersect(r, v);
-        int facet = P.num_of_hyperplanes();
+        int facet = (polypair.first < ball_lambda.first) ? polypair.second : P.num_of_hyperplanes();
 
-        if (polypair.first < ball_lambda.first ) facet = polypair.second;
+        return std::pair<NT, int>(std::min(polypair.first, ball_lambda.first), facet);
+    }
+
+    //---------------------aaccelerated billiard---------------------//
+    template <typename update_parameters>
+    std::pair<NT, int> line_first_positive_intersect(PointType const& r,
+                                                     PointType const& v,
+                                                     VT& Ar,
+                                                     VT& Av,
+                                                     update_parameters& params) const
+    {
+        std::pair <NT, int> polypair = P.line_first_positive_intersect(r, v, Ar, Av, params);
+        std::pair <NT, int> ball_lambda = B.line_positive_intersect(r, v);
+        
+        int facet = (polypair.first < ball_lambda.first) ? polypair.second : P.num_of_hyperplanes();
+        params.hit_ball = (polypair.first < ball_lambda.first) ? false : true;
```

Comment:
it is unnecessary to do both checks. 
what if
```
params.hit_ball = (polypair.first < ball_lambda.first) ? false : true;
int facet = params.hit_ball ? P.num_of_hyperplanes() : polypair.second;
```

Reply:
done